cmd/dist-receiver/main.go

```diff
@@ -0,0 +1,325 @@
+// Copyright © 2023 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	b64 "encoding/base64"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	config "github.com/ipfs/go-ipfs-config"
+	files "github.com/ipfs/go-ipfs-files"
+	"github.com/ipfs/go-ipfs/core"
+	"github.com/ipfs/go-ipfs/core/coreapi"
+	"github.com/ipfs/go-ipfs/core/node/libp2p"
+	"github.com/ipfs/go-ipfs/plugin/loader"
+	"github.com/ipfs/go-ipfs/repo/fsrepo"
+	icore "github.com/ipfs/interface-go-ipfs-core"
+	icorepath "github.com/ipfs/interface-go-ipfs-core/path"
+	"github.com/libp2p/go-libp2p-core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+var stageNow int
+var sig chan bool
+var targets []string
+
+func main() {
+	bootstrap := flag.String("bootstrap", "", "Specify the bootstrap node")
+	cidArg := flag.String("cid", "", "Specify the CID")
+	fileName := flag.String("file", "", "Specify the file name")
+	targetDir := flag.String("dir", "", "Specify the target directory")
+
+	flag.Parse()
+
+	stageNow = 0
+	sig = make(chan bool)
```

Comment:
maybe use chan struct{} is better when we don’t want to send a value over a channel but use it only as a way to signal an event.

FYI https://www.programming-books.io/essential/go/signaling-channel-with-chan-struct-f5daa999f5134b9ba9f2d69916df292a

Reply:
Updated to struct{}.